data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py

```diff
@@ -0,0 +1,134 @@
+"""
+Nullify concept ids for numeric PPIs from the RDR observation dataset
+
+Original Issues: DC-537, DC-703
+
+The intent is to null concept ids (value_source_concept_id, value_as_concept_id, value_source_value,
+value_as_string) for numeric PPIs from the RDR observation dataset. The changed records should be
+archived in the dataset sandbox.
+"""
+
+# Python imports
+import logging
+
+# Project imports
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.bq_utils import WRITE_TRUNCATE
+
+LOGGER = logging.getLogger(__name__)
+
+SAVE_TABLE_NAME = "dc_703_obs_changed_rows_saved"
+
+# Query to create tables in sandbox with the rows that will be removed per cleaning rule
+SANDBOX_QUERY = """
+CREATE OR REPLACE TABLE 
+    `{project}.{sandbox_dataset}.{intermediary_table}` AS (
+SELECT *
+FROM 
+    `{project}.{dataset}.observation`
+WHERE
+    questionnaire_response_id IS NOT NULL
+AND 
+    value_as_number IS NOT NULL
+AND 
+    (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL))
+"""
+
+CLEAN_NUMERIC_PPI_QUERY = """
+SELECT * REPLACE(
+    NULL AS value_source_concept_id,
+    NULL AS value_as_concept_id,
+    NULL AS value_source_value,
+    NULL AS value_as_string)
+FROM `{project}.{dataset}.observation` AS obs 
+WHERE 
+   questionnaire_response_id IS NOT NULL
+AND
+   value_as_number IS NOT NULL
+AND
+   (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL)
+"""
+
```

Comment:
When running this query I realized this is only selecting the 1m+ rows that are effected by the where clause. I asked in the eng channel about select all or select each column individually and Krishna suggested selecting each column individually, so I think the query below should work better:

SELECT
  observation_id,
  person_id,
  observation_concept_id,
  observation_date,
  observation_datetime,
  observation_type_concept_id,
  value_as_number,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_as_string,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_as_concept_id,
  qualifier_concept_id,
  unit_concept_id,
  provider_id,
  visit_occurrence_id,
  observation_source_value,
  observation_source_concept_id,
  unit_source_value,
  qualifier_source_value,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_source_concept_id,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_source_value,
  questionnaire_response_id
FROM
  `{project}.{dataset}.observation`